src/com/projectkorra/projectkorra/region/RegionProtection.java

```diff
@@ -176,14 +172,20 @@ private static boolean checkAll(Player player, Location location, CoreAbility ab
      */
     public static void startCleanCacheTask(double period) {
         Bukkit.getScheduler().runTaskTimer(ProjectKorra.plugin, () -> {
-            for (final Map<Block, BlockCacheElement> map : BLOCK_CACHE.values()) {
+            final long currentTime = System.currentTimeMillis();
```

Comment:
The value of a more technically accurate expiration time doesn't seem worth the trade-off checking the time for every cached block. Especially since this is would be a memory saving change the average memory use should go down overall with the other changes.